vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java

```diff
@@ -192,6 +193,21 @@ public void setLabelAsHtml(String htmlContent) {
         getElement().appendChild(labelElement.getElement());
     }
 
+    /**
+     * Replaces the label content with the given label component.
+     *
+     * @param component
+     *            the component to be added to the label.
+     *
+     * @since 23.1
+     */
+    public void setLabelComponent(Component component) {
```

Comment:
As this is a public API, we should also have ITs for it. Let's add at least some tests to `CheckboxIT` - as an example, you could copy and adapt some of the existing tests for `setLabelAsHtml()`.

Reply:
Added two tests: https://github.com/vaadin/flow-components/pull/3013/commits/1d3c3fb6e9e3085ce9f29ead0056892dec6eefe5